spring-integration-core/src/main/java/org/springframework/integration/config/xml/AbstractPollingInboundChannelAdapterParser.java

```diff
@@ -59,8 +59,8 @@ else if (source instanceof RuntimeBeanReference runtimeBeanReference) {
 		}
 		else {
 			parserContext.getReaderContext().error("Wrong 'source' type: must be 'BeanDefinition' or 'RuntimeBeanReference'", source);
+			throw new IllegalStateException("This dummy exception is meant to signify to  NullAway that an the error method throws an exception");
```

Comment:
I don't think it is correct and polite to say that as an exception message.
Just copy/paste from `error()` and be with it!
Plus, keep in mind the rule of not more than 120 symbols in line.